types/coin.go

```diff
@@ -410,6 +410,73 @@ func (coins Coins) SafeSub(coinsB ...Coin) (Coins, bool) {
 	return diff, diff.IsAnyNegative()
 }
 
+// Mul performs the scalar multiplication of coins with a `multiplier`
+// All coins are multipled by x
+// e.g.
+// {2A, 3B} * 2 = {4A, 6B}
+// {2A} * 0 panics
+// Note, if IsValid was true on Coins, IsValid stays true
+func (coins Coins) Mul(x Int) Coins {
+	coins, ok := coins.SafeMul(x)
+	if !ok {
+		panic("multiplying by zero is an invalid operation on coins")
+	}
+
+	return coins
+}
+
+// SafeMul performs the same arithmetic as Mul but returns false
+// if the `multiplier` is zero because it makes IsValid return false
+func (coins Coins) SafeMul(x Int) (Coins, bool) {
+	if x.IsNil() || x.IsZero() {
+		return nil, false
+	}
+
+	res := make(Coins, len(coins))
+	for i, coin := range coins {
+		c := coin
+		c.Amount = c.Amount.Mul(x)
+		res[i] = c
+	}
+
+	return res, true
+}
+
+// Quo performs the scalar division of coins with a `divisor`
+// All coins are divided by x and trucated.
+// e.g.
+// {2A, 30B} / 2 = {1A, 15B}
+// {2A} / 2 = {1A}
+// {4A} / {8A} = {0A}
+// {2A} / 0 = panics
+// Note, if IsValid was true on Coins, IsValid stays true,
+// unless the `divisor` is greater than the smallest coin amount.
```

Comment:
I am uncertain if we actually want this behavior, but I created these helpers function for avoiding having to do:

```go
sendCoins := coins
if split && !coins.IsZero() {
    sendCoins[0].Amount = coins[0].Amount.Quo(totalAddr)
}
```

If you find this better, then I can modify it and remove the changes in coins.